pommerman/network/ion_server/network.py

```diff
@@ -0,0 +1,205 @@
+                    f = load(f)
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_ok.value, f]), "utf8")))
+                else:
+                    await websocket.send(
+                        compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+        except:
+            await websocket.send(
+                compress(bytes(dumps([nic.status_fail.value]), "utf8")))
+    elif len(PLAYER_WS) < MAX_PLAYERS and (message["i"] == nic.match.value
+                                           or message["i"] == nic.room.value):
+        uuid = str(uuid4())
+        while uuid in PLAYER_WS:
+            uuid = str(uuid4())
+        PLAYER_WS[uuid] = {"ws": websocket}
+        if message["i"] == nic.match.value:
+            CONCURRENTLY_LOOKING["noroom"].append(uuid)
+            PLAYER_WS[uuid]["l"] = True
+        elif message["i"] == nic.room.value:
+            if message["room"] in CONCURRENTLY_LOOKING["room"]:
+                if len(CONCURRENTLY_LOOKING["room"][message["room"]]) < 4:
+                    CONCURRENTLY_LOOKING["room"][message["room"]].append(uuid)
+            else:
+                CONCURRENTLY_LOOKING["room"][message["room"]] = [uuid]
+            PLAYER_WS[uuid]["l"] = False
+            PLAYER_WS[uuid]["r"] = str(message["room"])
+        await websocket.send(dumps({"i": nic.status_reg.value, "id": uuid, "m": MODE}))
+
+
+async def ws_handler(websocket, pth=None): # pylint: disable=unused-argument
+    """Handle the messages recieved by WebSocket"""
+    global THREAD_LOCK
+    while THREAD_LOCK:
+        pass
+    THREAD_LOCK = True
+    async for message in websocket:
+        try:
+            message = loads(message)
+        except JSONDecodeError:
+            pass
+        try:
+            decode_message(message, websocket)
+        except:
+            pass
+        THREAD_LOCK = False
+        sleep(
+            0.0001
+        )  # Sleep for the a extremely small amount of time so the other thread can take the lock
+
+
+async def program_loop():
+    """Handles other network-related function"""
+    global THREAD_LOCK, CONCURRENTLY_LOOKING
+    while (True):
+        while THREAD_LOCK:
+            pass
+        THREAD_LOCK = True
+        for uuid in list(PLAYER_WS.keys()):
+            i = PLAYER_WS[uuid]
+            if i["ws"].closed:
+                if i["l"] is True:
+                    try:
+                        del CONCURRENTLY_LOOKING["noroom"][
+                            CONCURRENTLY_LOOKING["noroom"].index(uuid)]
+                    except:
+                        pass
+                elif i["l"] is False:
+                    try:
+                        del CONCURRENTLY_LOOKING["room"][i["r"]][CONCURRENTLY_LOOKING["room"][i["r"]].index(uuid)]
+                    except:
+                        pass
+                del PLAYER_WS[uuid]
+        if QUEUE_MAIN.poll():
```

Comment:
Woah, this came out of nowhere. I see it's defined down below. I'm goign to address it there.

Reply:
I want it to try deleting this even though the other one fails. Anyway, I've encased this in a `try-pass` as well.